### ISSUE TITLE
Add the concept of a "readiness" similar to k8s, etc.

### ISSUE THREAD

**Issue by alecthomas** [block/ftl#1294]
@mistermoe and I chatted about this

Could be something like this:

```go
//ftl:export
func Readiness(ctx context.Context) error {
	_, err := bearerDID.Get(ctx)
	return err
}
```

**Comment by deniseli** [comment]
Leaving some quick questions here for whoever ends up picking this up:

1. Should we require the function to be named `Readiness` or should we rely on an `//ftl:readiness` directive instead? If so, do we want `Readiness` to be a reserved name? i.e. no other functions can be named that.
1. Do we want to perform the readiness check (i.e. call each exported `Readiness` func) in `Service.Status()` in `controller.go`?
1. Should `readiness` effectively be a specialty verb? i.e. if a function is exported with the readiness directive, and only that directive, then it also automatically gets treated as a verb in all the ways a verb would be; it's just _also_ invoked for the readiness checks at startup? It's a bit odd because by the time the module is live, any Readiness function would presumably be returning `nil` persistently, so there wouldn't really be a point to calling it.
    1. Related: is there any reason to make these exportable? For module-to-module calls, it seems like exporting is just not useful. For calling it from the controller, we could argue that's a public/external access, in which case the visibility should technically be `public`, but that's basically implied already by the nature of `readiness`. 
    3. tl;dr: I'm leaning towards: `Readiness` is not a verb. It needs directive `//ftl:readiness` precisely. No explicit export in the directive. No module-to-module calls. The controller will call this function at startup.

**Comment by alecthomas** [comment]
I agree re. not being exported, I think `//ftl:export` in the example predated visibility controls.

I think it should be a verb for consistency though, as that is our unit of work for everything in FTL. I'm fine with either a well known name or `//ftl:readiness`, though the latter is more explicit.

**Comment by deniseli** [comment]
Sounds good! I'll pick this up after I finish my other issues if no one else beats me to it (but everyone definitely please feel free to take this!)
